pandas/core/indexes/multi.py

```diff
@@ -1391,6 +1391,25 @@ def format(
     # --------------------------------------------------------------------
     # Names Methods
 
+    def get_default_index_names(self, names=None):
+
+        if names is not None and not isinstance(names, (tuple, list)):
+            raise ValueError("Names must be a tuple or list")
+
+        if not names:
+            names = [
+                (n if n is not None else f"level_{i}") for i, n in enumerate(self.names)
+            ]
+        else:
+            if len(names) != self.nlevels:
```

Comment:
You can consider returning right away and getting rid of else statement, which will simplify this code:

``` python
if not names:
    return [...]

if len(names) != self.nlevels:
    raise ValueError

return names
```